app/src/main/java/com/example/myapplication/RegistrationActivity.kt

```diff
@@ -0,0 +1,86 @@
+package com.example.myapplication
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.activity.enableEdgeToEdge
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.ViewCompat
+import androidx.core.view.WindowInsetsCompat
+import com.google.android.material.button.MaterialButton
+import com.google.android.material.textfield.TextInputEditText
+import com.google.android.material.textfield.TextInputLayout
+
+class RegistrationActivity : AppCompatActivity() {
+
+    private val emailLayout: TextInputLayout
+        get() = findViewById(R.id.emailInputLayout)
+    private val emailEditText: TextInputEditText
+        get() = findViewById(R.id.emailEditText)
+    private val passwordLayout: TextInputLayout
+        get() = findViewById(R.id.passwordInputLayout)
+    private val passwordEditText: TextInputEditText
+        get() = findViewById(R.id.passwordEditText)
+    private val registerButton: MaterialButton
+        get() = findViewById(R.id.buttonNext)
+
+    private val credentialsManager = CredentialsManager()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        enableEdgeToEdge()
+        setContentView(R.layout.activity_register)
+        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
+            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
+            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
+            insets
+        }
+
+        registerButton.setOnClickListener {
+            val email = emailEditText.text.toString()
+            val password = passwordEditText.text.toString()
+
+            val isEmailValid = validateField(
+                layout = emailLayout,
+                value = email,
+                errorMessage = "Invalid email format"
+            ) { credentialsManager.isEmailValid(it) }
+
+            val isPasswordValid = validateField(
+                layout = passwordLayout,
+                value = password,
+                errorMessage = "Password must contain at least 8 characters, including uppercase, lowercase, number and special character"
```

Comment:
suggestion(non-blocking): Yep, same as the comment above. Same string, different place. Looks like a nice candidate for a resource string :D